tools/harness-simulation/posix/sniffer_sim/sniffer.py

```diff
@@ -28,104 +28,172 @@
 #
 
 import argparse
+from concurrent import futures
+import enum
+import grpc
+import ipaddress
 import logging
 import signal
-import time
 import pcap_codec
-import sys
 import threading
 
+from proto import sniffer_pb2
+from proto import sniffer_pb2_grpc
 import sniffer_transport
 
 
-class Sniffer:
-    """ Class representing the Sniffing node, whose main task is listening.
-    """
+class SnifferServicer(sniffer_pb2_grpc.Sniffer):
+    """ Class representing the Sniffing node, whose main task is listening. """
 
-    logger = logging.getLogger('sniffer.Sniffer')
+    logger = logging.getLogger('sniffer.SnifferServicer')
 
     RECV_BUFFER_SIZE = 4096
+    MAX_NODES_NUM = 33
 
-    def __init__(self, filename, channel):
-        self._pcap = pcap_codec.PcapCodec(filename, channel)
-
-        # Create transport
-        transport_factory = sniffer_transport.SnifferTransportFactory()
-        self._transport = transport_factory.create_transport()
+    class State(enum.Enum):
+        STOPPED = 0
+        RUNNING = 1
 
+    def _clear(self):
```

Comment:
```suggestion
    def _reset(self):
```
`reset` might be a better name?

Reply:
Accepted.